Add typescript informations in coding standards (#1083)
PrestaShop/docs: development/coding-standards/_index.md

You can run the linter to help you comply with these coding standards:

## TypeScript

> Most of `admin-dev/themes/new-theme/js` files are coded in TypeScript. Classes and functions in .ts files must be strictly typed. 

Reviewer: ```suggestion
All `admin-dev/themes/new-theme/js` files are coded in TypeScript. Classes and functions in .ts files must be strictly typed. 
```

npm run scss-fix
```

> ## TypeScript

Reviewer 2: Can you please move this block under the Javascript section? This is more like a third-level block.

> Most of `admin-dev/themes/new-theme/js` files are using TypeScript. This provide a better maintainability by using types on most of our classes and functions. 

Reviewer 2: ```suggestion

Most of `admin-dev/themes/new-theme/js` files are coded in TypeScript. Classes and functions in .ts files must be strictly typed. 
```

You are able to get global types in the `admin-dev/themes/new-theme/js/types` folder and some types library are imported from npm using the `@types` namespace and automatically imported by TypeScript.

> The `tsconfig.json` and `webpack` configuration files are adding a `@PSTypes` relative path to the `admin-dev/themes/new-theme/js/types` folder, because `@types` is a reserved folder.

Reviewer 2: I'm struggling to understand this part. It doesn't look like coding standards to me.

Author: We should tell peoples that our types are in the @PSTypes relative path, and external types are in the @types namespace, but @types is a standard so we are maybe not forced to explain it as it's TS standard
